Will not compile with -fno-exceptions

I am working on a bare metal ARM-cortex M project using this units library.

I just updated to the head of your main branch and changes in `src/core/include/units/magnitude.h` are causing compilation errors. Specifically the use of `throw` statements. I have it working now by guarding each throw with the macro
```
#if __has_feature(cxx_exceptions) || defined(__cpp_exceptions) || \
    (defined(_MSC_VER) && defined(_CPPUNWIND)) || \
    defined(__EXCEPTIONS)
```

EDIT:
I think this may not be related to `-fno-exceptions`, since it does compile if I make some small code changes. I am trying to fins the root cause and will update/close issue when I find out more (see my comments below).

Compiler: arm-none-eabi-11.3.1-rel1 (latest ARM provided toolchain)
arch: ARM-v8M-baseline (M23)

Changing code to the following: gave a much more readable compile error(s):
```
value_type do_value() noexcept {
    using namespace units::isq::si::references;
    const uint32_t reading = hal_read_battery();
    uint32_t millivolts = (reading / ( max_adc_reading * battery_ratio)) * std::milli::den;
    return value_type {millivolts * mV};
}
```

with the line that stuck out to me being (cast with _long unsigned int_ and _long int_). The exception hit isn't super clear to me but sounds like it's trying to prevent a narrowing conversion?
```
In file included from E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/bits/basic_concepts.h:31,
                 from E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/concepts.h:26,
                 from E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/systems/isq/include/units/isq/dimensions/voltage.h:25,
                 from E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/systems/si/include/units/isq/si/voltage.h:26,
                 from E:/wildlife/nextgen/mintag-simulator/seal/src/seal/sensors/voltage.hpp:4,
                 from E:/wildlife/nextgen/mintag-simulator/saml1x-feather/src/platform.hpp:6,
                 from E:/wildlife/nextgen/mintag-simulator/saml1x-feather/src/platform.cpp:1:
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/magnitude.h: In instantiation of 'constexpr To units::detail::checked_static_cast(From) [with To = long long int; From = long long int]':
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/magnitude.h:392:57:   required from 'constexpr T units::get_value(const units::magnitude<BPs ...>&) [with T = long long int; auto ...BPs = {}]'
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/quantity_cast.h:113:75:   required from 'units::quantity_cast<units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long int>, units::isq::si::dim_voltage, units::isq::si::millivolt, long unsigned int>(const units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long unsigned int>&)::<lambda(auto:42)> [with auto:42 = units::magnitude<>]'
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/quantity_cast.h:114:82:   required from 'constexpr auto units::quantity_cast(const units::quantity<D, U, Rep>&) [with To = units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long int>; D = units::isq::si::dim_voltage; U = units::isq::si::millivolt; Rep = long unsigned int]'
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/quantity.h:147:83:   required from 'constexpr units::quantity<D, U, Rep>::quantity(const Q&) [with Q = units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long unsigned int>; D = units::isq::si::dim_voltage; U = units::isq::si::millivolt; Rep = long int]'
E:/wildlife/nextgen/mintag-simulator/saml1x-feather/src/platform.cpp:32:37:   required from 'seal::Result<units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long int> > MyVoltageSensor<bits>::do_value() [with unsigned int bits = 12; seal::Result<units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long int> > = outcome_v2::basic_result<units::quantity<units::isq::si::dim_voltage, units::isq::si::millivolt, long int>, system_error2::errored_status_code<system_error2::erased<int> >, seal::ResultViolationPolicy>]'
E:/wildlife/nextgen/mintag-simulator/saml1x-feather/src/platform.cpp:28:51:   required from here
E:/wildlife/nextgen/mintag-simulator/seal/external/mp-units/core/include/units/magnitude.h:221:7: error: exception handling disabled, use '-fexceptions' to enable
  221 |       throw std::invalid_argument{"Cannot represent magnitude in this type"};
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Changing code to do the narrowing cast to int32_t before making it into a unit also fixes the code and seems "correct" to me:
```
    seal::Result<seal::VoltageSensor::value_type> do_value() noexcept override {
        using namespace units::isq::si::references;
        const uint32_t reading = board::analog::read_battery();
        const int32_t millivolts = (reading / ( max_value() * voltage_ratio())) * std::milli::den;
        return millivolts * mV;
    }
```

--- a/src/core/include/units/magnitude.h
+++ b/src/core/include/units/magnitude.h
@@ -27,9 +27,9 @@
 #include <units/ratio.h>
 #include <concepts>
 #include <cstdint>
+#include <exception>
 #include <numbers>
 #include <optional>
-#include <stdexcept>
 
 namespace units {
 namespace detail {
@@ -147,10 +147,10 @@ using widen_t =
 template<typename T>
 constexpr T int_power(T base, std::integral auto exp)
 {
-  // As this function should only be called at compile time, the exceptions herein function as
-  // "parameter-compatible static_asserts", and should not result in exceptions at runtime.
+  // As this function should only be called at compile time, the terminations herein function as
+  // "parameter-compatible static_asserts", and should not result in terminations at runtime.
   if (exp < 0) {
-    throw std::invalid_argument{"int_power only supports positive integer powers"};
+    std::terminate();  // int_power only supports positive integer powers
   }
 
   constexpr auto checked_multiply = [](auto a, auto b) {
@@ -158,7 +158,7 @@ constexpr T int_power(T base, std::integral auto exp)
     UNITS_DIAGNOSTIC_PUSH
     UNITS_DIAGNOSTIC_IGNORE_FLOAT_EQUAL
     if (result / a != b) {
-      throw std::overflow_error{"Wraparound detected"};
+      std::terminate();  // Wraparound detected
     }
     UNITS_DIAGNOSTIC_POP
     return result;
@@ -188,14 +188,14 @@ constexpr widen_t<T> compute_base_power(BasePower auto bp)
   // need to write a custom function.
   //
   // Note that since this function should only be called at compile time, the point of these
-  // exceptions is to act as "static_assert substitutes", not to throw actual exceptions at runtime.
+  // terminations is to act as "static_assert substitutes", not to actually terminate at runtime.
   if (bp.power.den != 1) {
-    throw std::invalid_argument{"Rational powers not yet supported"};
+    std::terminate();  // Rational powers not yet supported
   }
 
   if (bp.power.num < 0) {
     if constexpr (std::is_integral_v<T>) {
-      throw std::invalid_argument{"Cannot represent reciprocal as integer"};
+      std::terminate();  // Cannot represent reciprocal as integer
     } else {
       return T{1} / compute_base_power<T>(inverse(bp));
     }
@@ -214,11 +214,11 @@ template<typename To, typename From>
   requires(!std::is_integral_v<To> || std::is_integral_v<From>)
 constexpr To checked_static_cast(From x)
 {
-  // This function should only ever be called at compile time.  The purpose of these exceptions is
+  // This function should only ever be called at compile time.  The purpose of these terminations is
   // to produce compiler errors, because we cannot `static_assert` on function arguments.
   if constexpr (std::is_integral_v<To>) {
     if (!std::in_range<To>(x)) {
-      throw std::invalid_argument{"Cannot represent magnitude in this type"};
+      std::terminate();  // Cannot represent magnitude in this type
     }
   }
 

